beancount/utils/errors.cc

```diff
@@ -22,13 +23,20 @@ absl::StatusCode kSystemError = absl::StatusCode(kUserErrors + 1);
 
 absl::Status SystemError(absl::string_view message) {
   char buffer[2048];
-#ifndef __APPLE__
-  description = strerror_r(errno, buffer, 2048);
+
+#ifdef __GLIBC__
+  // GNU libc provides non-XSI-compliant strerror_r.
+  int status = errno;
+  errno = 0;
+  strerror_r(status, buffer, 2048);
+  status = errno;
 #else
   int status = strerror_r(errno, buffer, 2048);
-  assert(status == 0);
 #endif
 
+  // TODO: Handle errors raised by strerror_r correctly.
+  assert(status == 0);
+
```

Comment:
The right way to fix this is a configure time check for the `strerror_r()` function. I'm sure Bazel should have an easy way to do it. However, what's the point to trade a compilation error with an assertion failure at runtime?

Reply:
Thank you for reviewing!

The assertion here, derived from the present code, checks if strerror_r itself fails run-time. For example, the function has no message string for given errno for some reasons.

The proposed code, however, is not ideal as no precise error message does not mean it is a fatal error. returning an error indicating a failure to retrieve an appropriate error message may be better. Your opinion?

Reply:
Sorry, I misread the code.

Reply:
Added a proper error handling.